src/app/video-courses/components/video-courses-list/video-courses-list.component.ts

```diff
@@ -48,33 +57,36 @@ export class VideoCoursesListComponent implements OnInit, OnDestroy {
   }
 
   public search(searchCriteria: string) {
-    this.router.navigate([], {
-      relativeTo: this.route,
-      queryParams: {
-        search: searchCriteria || null,
-      },
-      queryParamsHandling: 'merge',
-    });
+    this.searchSubject.next(searchCriteria);
   }
 
   public loadMore() {
+    this.loadingService.setLoading(true);
     const getSub = this.videoCoursesService
       .getCourses(this.courses.length, this.coursesChunkSize)
       .subscribe((courses) => {
         if (courses.length < this.coursesChunkSize) {
           this.isFullyLoaded = true;
         }
         this.courses = [...this.courses, ...courses];
+        this.loadingService.setLoading(false);
       });
     this.subs.add(getSub);
   }
 
+  private serachVideoCourses(searchCriteria: string) {
+    const sub = this.videoCoursesService.getCourses(0, this.coursesChunkSize, searchCriteria).subscribe((courses) => {
+      this.courses = courses;
+    });
+    this.subs.add(sub);
+  }
+
   private getVideoCourses() {
-    const sub = this.route.queryParams
-      .pipe(switchMap((params: Params) => this.videoCoursesService.getCourses(0, this.coursesChunkSize, params.search)))
-      .subscribe((courses) => {
-        this.courses = courses;
-      });
+    this.loadingService.setLoading(true);
+    const sub = this.videoCoursesService.getCourses(0, this.coursesChunkSize).subscribe((courses) => {
```

Comment:
Could you please also check topic **subscribe() vs | async Pipe** ( first found link, but maybe you will find better explanations https://medium.com/angular-in-depth/angular-question-rxjs-subscribe-vs-async-pipe-in-component-templates-c956c8c0c794)  Async pipes Unsubscribe automatically, very often it is worth to use them instead of .subscribe()